File: Learnify.Domain/Entities/Leaderboard.cs
namespace Learnify.Domain.Entities
{
    public class Leaderboard : BaseEntity
    {
        public double Points { get; set; }
        public int Rank { get; set; }
        public Guid StudentId { get; set; }
        public User? Student { get; set; }
    }
}

File: Learnify.UI/src/models/entities/Leaderboard.js
import { BaseEntity } from "./BaseEntity";

export class Leaderboard extends BaseEntity {
    constructor(id, points, rank, studentId, student = null) {
        super(id);
        this.points = points;
        this.rank = rank;
        this.studentId = studentId;
        this.student = student;
    }
}